### ISSUE TITLE
`type` is not selectively disclosable

### ISSUE THREAD

**Issue by awoie** [oauth-wg/oauth-sd-jwt-vc#11]
Currently, `type` is not one of the selectively disclosable claims. This might be ok but it might have some privacy implications if a VC has multiple types. Let's assume the following:

A VC with the types: driver's-license, age-over-18.

If the holder discloses only the age, then based on the current specification, the verifier would also know that the age was disclosed from a DL which is probably unnecessary context info since all the verifier needs is to trust the issuer of the VC. Note, the issuer itself might also leak some contextual info as well, so the issue might not need to be mitigated or it might not be entirely possible anyways.

Should we make `type` selectively disclosable per array entry? Is this even supported by SD-JWT?

**Comment by awoie** [comment]
We might want to add something to privacy considerations about that.

**Comment by TakahikoKawasaki** [comment]
FYI: There is a discussion about Selective Disclosure for array elements.

https://github.com/oauth-wg/oauth-selective-disclosure-jwt/issues/194#issuecomment-1487860648

**Comment by tlodderstedt** [comment]
suggest to close as the type is a simple string now

**Comment by Sakurann** [comment]
@peteraltmann raised an issue with me that "if the type is always disclosed, what is the point of selective disclosure because the verifier will know what claims are in the credential". I think we should consider making type selectively disclosable.

**Comment by peppelinux** [comment]
I agree with Peter and I'm in favor of sd for type

**Comment by tlodderstedt** [comment]
What is the value of a VC without type information? Or let me put it the other way around. If the Verifier is supposed to be able to process a VC without knowing the type (which also means it can request the VC presentation without specifying the type), why do we need the type at all?

**Comment by AltmannPeter** [comment]
Having the type disclosable does not mean the user never shares it. If a verifier requests the type then the user can disclose it, no? 

The issues with both the @context and the type is that they reveal what claims the user has, which is more than what is strictly required for a service. The privacy requirements from the Parliament, Council, and the Commission are strict. However, I cannot think of a scenario to justify type being disclosable.

Arguments for: As @awoie mentions, the verifier may require only a particular claim, e.g., age over 18, and should reasonable accept these from any trusted issuer without having to know what exact attestation type it came from. Some issuers may have multiple types that all contain the same attribute. And there are strong demands that the user should reveal only what is required for the service and absolutely nothing else (to the point where I am actively having to fight against people who want to put ZKP everywhere). Not revealing claim values is great; also hiding what claims you have may be desirable.

Arguments against: Beside from those already mentioned, you could question the privacy benefits. As long as the individual Issuer's signature is included, it is likely not that hard to guess what the particular attestation may contain, particularly if the Issuer only issues few attestations.

In conclusion, given how there are no technical challenges to making type disclosable, and how the Verifier may just request a type and refuse to process a presented claim without type disclosed, I would argue for that the type MAY be disclosed.

**Comment by paulbastian** [comment]
The `iss` field is always disclosed, hence not revealing the `type` only offers privacy if the same issuer with the same key offers multiple VCs that share the same claim name that is being requested. It seems like an unrealistic scenario that this is a privacy benefit. It seems to me the usecase would be merely a fillout help, where the verifier does not care about the origin of claims and handles those as self-attested

**Comment by awoie** [comment]
Let's create a PR and see how the spec will change and continue the discussion there.

**Comment by awoie** [comment]
Currently, the spec gives `type` the following function:
> A type defines which claims may or must appear in the Unsecured Payload of the SD-JWT VC and whether they may, must, or must not be selectively disclosable.

Given that, I don't think it can be selectively disclosed. We should close this issue.

**Comment by awoie** [comment]
We should add something to the privacy considerations section that explains the risk of leaking more context info from the type in certain use cases.
